pgmpy/metrics/metrics.py

```diff
@@ -465,3 +465,189 @@ def SHD(true_model, est_model):
+    >>> from pgmpy.factors.discrete import TabularCPD
+    >>> from pgmpy.sampling import BayesianModelSampling
+    >>> from pgmpy.estimators import HillClimbSearch
+    >>> from pgmpy.metrics import self_compatibility_graphical
+    >>> # 1) Build a simple BN X→Y
+    >>> model = DiscreteBayesianNetwork([("X", "Y")])
+    >>> model.add_cpds(
+    ...     TabularCPD("X", 2, [[0.5], [0.5]]),
+    ...     TabularCPD("Y", 2,
+    ...                [[0.8, 0.2],
+    ...                 [0.2, 0.8]],
+    ...                evidence=["X"], evidence_card=[2])
+    ... )
+    >>> # 2) Sample 100 rows
+    >>> df = BayesianModelSampling(model).forward_sample(size=100)
+    >>> # 3) Compute graphical self-compatibility
+    >>> score = self_compatibility_graphical(
+    ...     HillClimbSearch,
+    ...     df,
+    ...     num_subsets=3,
+    ...     subset_fraction=0.5,
+    ...     scoring_method="bic-d"
+    ... )
+    >>> isinstance(score, float)
+    True
+
+    References
+    ----------
+    [1] Faller, P. M., et al. (2024).
+        “Self‐compatibility: Evaluating Causal Discovery without Ground Truth.”
+        In AISTATS. arXiv:2307.09552
+    """
+    rng = np.random.RandomState(random_state)
+    variables = list(data.columns)
+    p = len(variables)
+    k = max(2, int(np.floor(subset_fraction * p)))
+
+    def _latent_admg(dag: DAG, observed: list) -> nx.DiGraph:
+        """
+        Compute the latent‐projection ADMG L(G, S) of a DAG G onto a subset S ⊂ V.
+
+        Implements Definition 5 (latent ADMG) from [1] (Faller et al., AISTATS 2024):
+        Let G be an ADMG with variables V and S ⊂ V . The latent ADMG L(G, S)
+        is the ADMG that contains all nodes in S,
+        all edges between nodes in S and additionally
+        1) a directed edge between X, Y ∈ S if there is a directed path from X to Y where all intermediate
+            nodes are in V \ S
+        2) a bidirected edge between X, Y if there is a (undirected) path such that every non-endpoint is a noncollider in V \ S and there are arrowheads towards
+            X and Y on the incident edges on the path.
+
+        Parameters
+        ----------
+        dag : DAG
+            The full DAG G on variables V (may include latent nodes).
+        observed : list
+            Subset S ⊂ V to project onto (observed variables).
+
+        Returns
+        -------
+        nx.DiGraph
+        An ADMG over the observed nodes S, encoded as a NetworkX DiGraph where:
+          - A single arc u→v indicates a latent‐only directed chain from u to v.
+          - A bidirected link u↔v is encoded by having both arcs u→v and v→u.
+
+        References
+        ----------
+        [1] Faller, P. M., et al. (2024).
+        “Self‐compatibility: Evaluating Causal Discovery without Ground Truth.”
+        In AISTATS. arXiv:2307.09552
+        """
+        # 1) Prep
+        full_nx = dag.to_directed()
+        V = set(dag.nodes())
```

Comment:
Try to use more readable variable names, such as `variables`, `vars`, or `nodes`